File: CCLib/src/java/lang/native/PrintStream.lua
natives["java.io.PrintStream"] = natives["java.io.PrintStream"] or {}

local function lprint(s, p)
	if s ~= nil then
		p(s)
	else
		p("(null)")
	end
end

natives["java.io.PrintStream"]["println(Ljava/lang/String;)V"] = function(this, str)
	lprint(str, this.fields.err.value ~= 0 and printError or print)
end

natives["java.io.PrintStream"]["println(Z)V"] = function(this, str)
	lprint(str == 0, this.fields.err.value ~= 0 and printError or print)
end

natives["java.io.PrintStream"]["println(I)V"] = function(this, str)
	lprint(str, this.fields.err.value ~= 0 and printError or print)
end

File: CCLib/src/java/io/PrintStream.java
package java.io;

public class PrintStream {
	private boolean err;

	public native void println(String in);
	public native void println(int in);
	public native void println(boolean in);

	public PrintStream() {
		this(false);
	}

	public PrintStream(boolean err) {
		this.err = err;
	}
}